### ISSUE TITLE
New Server API Phase 2

### ISSUE THREAD

**Issue by purplefox** [confluentinc/ksql#4266]
This is an umbrella task which tracks all the issues for phase 2 of implementing KLIP-15 - new Server API.

It should not be started until https://github.com/confluentinc/ksql/issues/4254 is complete.

Phase 2 is more or less all the remaining tasks required after phase 1 to complete KLIP-15.

It includes:

- [x] - [Migrate status endpoint to new API](https://github.com/confluentinc/ksql/issues/4280)
- [x] - [Migrate statement execution endpoint to new API](https://github.com/confluentinc/ksql/issues/4281)
- [x] - [Migrate terminate endpoint to new API](https://github.com/confluentinc/ksql/issues/4282)
- [x] - [Migrate healthcheck endpoint to new API](https://github.com/confluentinc/ksql/issues/4290)
- [x] - [Migrate info endpoint to new API](https://github.com/confluentinc/ksql/issues/4291)
- [x] - [Migrate metadata endpoint to new API](https://github.com/confluentinc/ksql/issues/4292)
- [x] - [Refactor the KsqlClient and related classes to use Vert.x](https://github.com/confluentinc/ksql/issues/4283)
- [x] - [Update C3 to work with new API](https://github.com/confluentinc/ksql/issues/4284)
- [x] - [Update Jetty specific plugins to work with new API](https://github.com/confluentinc/ksql/issues/4285)
- [x] - [Implement Java client](https://github.com/confluentinc/ksql/issues/4265)
- [x] - [Consider CORS support](https://github.com/confluentinc/ksql/issues/4307)
- [x] - [Limits on push queries](https://github.com/confluentinc/ksql/issues/4475)
- [x] - [Websockets streaming endpint](https://github.com/confluentinc/ksql/issues/4476)
- [x] - [Better error message when HTTP/1.1 is used with new API server](https://github.com/confluentinc/ksql/issues/4604)
- [x] - [Cluster health check /chc equivalent](https://github.com/confluentinc/ksql/issues/5077)
- [x] - [Unify API error codes](https://github.com/confluentinc/ksql/issues/5249)
- [x] - [Metrics for API endpoints](https://github.com/confluentinc/ksql/issues/5250)
- [x] - [Integrate metrics into datadog](https://github.com/confluentinc/ksql/issues/5372)

**Comment by derekjn** [comment]
@purplefox @colinhicks should we split this into smaller, incremental issues? It seems like this issue may represent more work than we can get done in a single release.

**Comment by colinhicks** [comment]
@derekjn, the incremental child issues are enumerated in the description above.

@purplefox @vcrfxia, for some context, we are looking at the 0.9.0 milestone. We'd like to add the appropriate child issues to the milestone with your guidance on what's possible to complete by the next release.

**Comment by derekjn** [comment]
@purplefox @vcrfxia can you guys label the smaller issues here that we can fit into `0.9.0` as P0's? Once we have those we can remove this issue from `0.9.0`.

**Comment by purplefox** [comment]
Hi folks, when is 0.9.0 due?

**Comment by vcrfxia** [comment]
Removed https://github.com/confluentinc/ksql/issues/4286 from this epic, as it will be pushed into 0.11.0 instead of 0.10.0. Marking this epic as complete.

Remaining Java client tasks are tracked separately:
https://github.com/confluentinc/ksql/issues/5583
https://github.com/confluentinc/ksql/issues/5584
https://github.com/confluentinc/ksql/issues/4286
